ert3/evaluator/__init__.py

```diff
@@ -1 +1 @@
-from ert3.evaluator._evaluator import evaluate
+from ert3.evaluator._evaluator import evaluate, get_evaluator_tmp_dir
```

Comment:
I'd prefer this function to not be exposed, as it suddenly leaks the file system usage... How about if we instead change ert3 to:
- delete the experiment `tmp_dir` to be used if it already exists and
- delete the experiment `tmp_dir` if the experiment succeeds.

Now, the later will have to be done in `engine` as of now since the data first needs to be persisted in `storage`. So maybe the evaluator will need to expose something like a `clean_experiment_cache`-function to do this. If so, we might also just use that function when cleaning the experiment instead. Which would result in the behaviour as currently achieved, but with the knowledge of the tmp_dir entirely contained within the `evaluator`.